tests/js/core/widget/core/Spin/Spin.js

```diff
@@ -1,181 +1,352 @@
+
+			spinWidget.value(testValues[0]);
+			equal(spinWidget.value(), testValues[0], "Spin value check (0)");
+			spinWidget.value(testValues[1]);
+			equal(spinWidget.value(), testValues[1], "Spin value check (10)");
+			spinWidget.value(testValues[2]);
+			equal(spinWidget.value(), spinWidget.option("max"), "Spin value check (100) limited to max");
+			spinWidget.option("loop", "enabled");
+			spinWidget.value(testValues[3]);
+			equal(spinWidget.value(), spinWidget.option("min"), "Spin value check (max + 1) equal min for loop enabled");
+
+			spinWidget.destroy();
+		});
+
+		test("Test of _updateItems", 3, function (assert) {
+			var widget = new Spin();
+
+			helpers.stub(widget, "_removeSelectedLayout", function () {
+				assert.ok(true, "_removeSelectedLayout has been called");
+				return false;
+			});
+
+			helpers.stub(widget, "_modifyItems", function () {
+				assert.ok(true, "_modifyItems has been called");
+			});
+
+			helpers.stub(widget, "_addSelectedLayout", function () {
+				assert.ok(true, "_addSelectedLayout has been called");
+			});
+
+			widget._updateItems();
+		});
+
+		test("Test of _setMax", 2, function (assert) {
+			var widget = new Spin(),
+				element;
+
+			widget.options.max = 10;
+			widget.options.min = 0;
+			widget.length = 10;
+			widget.step = 1;
+
+			widget._setMax(element, 100);
+
+			assert.equal(widget.options.max, 100, "Max 100 has been set");
+			assert.equal(widget.length, 101, "Length of items has been changed to 101");
+		});
+
+		test("Test of _setMin", 2, function (assert) {
+			var widget = new Spin(),
+				element;
+
+			widget.options.min = 10;
+			widget.options.max = 20;
+			widget.length = 11;
+			widget.step = 1;
+
+			widget._setMin(element, 0);
+
+			assert.equal(widget.options.min, 0, "Min 100 has been set");
+			assert.equal(widget.length, 21, "Lenght of items has been changed to 21");
+		});
+
+		test("Test of _setEnabled", 9, function (assert) {
+			var widget = new Spin(),
+				element = document.getElementById("spin");
+
+			widget.element = element;
+			widget.options.enabled = false;
+			widget.dragTarget = element;
+			widget._animation = {
+				stop: function () {
+					assert.ok(true, "_animation.stop() has been called");
+				}
+			}
+			helpers.stub(window, "setTimeout", function () {
+				assert.ok(true, "setTimeout has been called");
+			});
+
+			// ENABLED
+			widget._setEnabled(element, true);
+
+			assert.ok(widget.options.enabled, "Widget enabled");
+			assert.ok(element.classList.contains("ui-spin-enabling"), "Widget element has class 'ui-spin-enabling'");
+			assert.ok(element.classList.contains("enabled"), "Widget element has class 'enabled'");
+
+			// DISABLE
+			widget.options.enabled = true;
+
+			widget._setEnabled(element, false);
+
+			assert.ok(!widget.options.enabled, "Widget enabled");
+			assert.ok(element.classList.contains("ui-spin-enabling"), "Widget element has class 'ui-spin-enabling'");
```

Comment:
Is it correct to have the "ui-spin-enabling" when the element is disabled?

Reply:
Yes. This is correct. This is a class that indicates that the widget is switching. Adding / removing it in the test due to implementations